File: App/Resources/AlgorithmDetails/shovesort/shovesort.swift
func sort(_ arr: inout [Int]) {
    let end = arr.count
    var i = 0
    while i < end - 1 {
        if arr[i] > arr[i + 1] {
            for f in i..<(end - 1) {
                arr.swapAt(f, f + 1)
            }
            if i > 0 {
                i -= 1
            }
            continue
        }
        i += 1
    }
}

var array: [Int] = [
    0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56,
]
sort(&array)
print(array)

File: App/Resources/AlgorithmDetails/shovesort/shovesort.go
package main

import (
	"fmt"
)

func sort(arr []int) []int {
	end := len(arr)
	i := 0
	for i < end-1 {
		if arr[i] > arr[i+1] {
			for f := i; f < end-1; f++ {
				arr[f], arr[f+1] = arr[f+1], arr[f]
			}
			if i > 0 {
				i--
			}
			continue
		}
		i++
	}
	return arr
}

func main() {
	array := []int{0, 39, 21, 62, 91, 77, 14, 23,
		90, 69, 51, 81, 68, 83, 32, 56}
	fmt.Println(sort(array))
}
